### PR TITLE
Dotnet upgrade fix

### PR DESCRIPTION
**Pull request by hsubramanianaks**
Thanks @elenavillamil heaps for suggesting the fix. Thank you @pkoelbl for creating this PR. 

This PR from the main [PR](https://github.com/Azure/Bridge-To-Kubernetes/pull/48) plus the following fixes 

- for control port stop [issue](https://github.com/Azure/Bridge-To-Kubernetes/issues/98)
- routing manager launch [issue](https://github.com/Azure/Bridge-To-Kubernetes/issues/112)
- endpoint manager launch [issue](https://github.com/Azure/Bridge-To-Kubernetes/issues/99)

Thanks everyone who have checked this PR.

### CODE REVIEW

File: samples/ingress/database-api/.vscode/launch.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
             "request": "launch",
             "preLaunchTask": "build",
             // If you have changed target frameworks, make sure to update the program path.
```
Comment (Tatsinnit) [inline samples/ingress/database-api/.vscode/launch.json:12]: 💡 I wonder comments like this have any historical background that we can keep them for those reasons?

**Review by Tatsinnit — DISMISSED** [review]
🚀☕️ Thank you for adding me in this PR, looks good to me, given its a progressive PR if test pass and build looks good, I think best review could be test as this is a high impact large volume of files PR.

Might be folks who can test can come super handy for green-lighting. Thanks heaps. :shipit:

File: build/setup-kubectl.sh [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+        loc_kube_version=$DEFAULT_KUBECTL_VERSION
+    fi
+
+    if [ -z "$loc_setup_location" ]; then
+        loc_setup_location=$DEFAULT_INSTALL_LOCATION
+    fi
 
-    apk add -u kubectl --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing
+    echo "Setting up kubectl $loc_kube_version in $loc_setup_location"
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/${loc_kube_version}/bin/linux/amd64/kubectl
```
Comment (pkoelbl) [inline build/setup-kubectl.sh:21]: This will cause a problem if you are using ARM64 because it is only install the amd64 version

Comment (hsubramanianaks) [reply build/setup-kubectl.sh:21]: @pkoelbl we tested in ARM64 Mac M1 max , we didn't face any errors/issues for both with and without routing. Tired couple of times connecting and disconnecting, it worked fine. Can you test this PR and provide the logs ? here is the CLI Version 1.0.20221129.4 and set BRIDGE_ENVIRONMENT='dev' in your environment variables. Thanks.

Comment (hsubramanianaks) [reply build/setup-kubectl.sh:21]: Once you confirm we will proceed further, even with our testing team we didn't see any issues with M1 Mac's. I am wondering if I am missing something. Thanks.

File: samples/ingress/database-api/.vscode/launch.json (hunk 1 above)
Comment (hsubramanianaks) [reply samples/ingress/database-api/.vscode/launch.json:12]: @Tatsinnit Yes we are keeping these, it is just that GitHub displays them in a weird way as if it is deleted.

File: src/.vscode/launch.json [hunk 3]
```diff
@@ -18,9 +18,9 @@
             "preLaunchTask": "BuildAndPublishMac", // Replace by BuildAndPublishWindows as needed
             // If you have changed target frameworks, make sure to update the program path.
             // Replace osx-x64 with win-x64 if you are running in windows
-            "program": "${workspaceFolder}/dsc/bin/Debug/netcoreapp3.1/osx-x64/dsc.dll",
-            "args": ["connect", "--service", "mi-webapp-service", "--local-port", "3000", "--namespace", "mi-webapp2", "--use-kubernetes-service-environment-variables"],
-            //"args": ["prep-connect", "--service", "stats-api"],// "--local-port", "3001", ],
+            "program": "${workspaceFolder}/dsc/bin/Debug/net6.0/win-x64/dsc.dll",
```
Comment (elenavillamil) [inline src/.vscode/launch.json:21]: preLaunchTask has BuildAnPublishMac, but path here has win, maybe update it to osx for consistency with the prelaunch task?

**Review by sabbour — APPROVED** [review]
Changes look good to me, thanks for the great effort! Looking forward to the testing in staging.

**Review by bfoley13 — APPROVED** [review]
changes look good

File: build/setup-kubectl.sh (hunk 2 above)
Comment (hsubramanianaks) [reply build/setup-kubectl.sh]: @pkoelbl any update on this ? At this moment this PR is mature enough to be merged into main. Please let us know. Thanks.

**Review by elenavillamil — APPROVED** [review]
Awesome work!

**Review by elenavillamil — APPROVED** [review]

File: build/setup-alpine-tools.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/bin/ash
+#!/bin/bash
```
Comment (Tatsinnit) [inline build/setup-alpine-tools.sh:1]: ☕️💡question please - why ash (default shell) is dropped and bash is used? Ash is minimalist she’ll also has some good security by design advantage. 

* https://wiki.alpinelinux.org/wiki/Change_default_shell

File: samples/todo-app/stats-api/.vscode/tasks.json [hunk 5]
```diff
@@ -9,10 +9,10 @@
             "ports": [
                 3001
             ],
-            "targetCluster": "bridgeTesting-0812",
+            "targetCluster": "hari-aks",
             "targetNamespace": "todo-app",
             "useKubernetesServiceEnvironmentVariables": false,
-            "isolateAs": "elvilla-7f7e"
```
Comment (Tatsinnit) [inline samples/todo-app/stats-api/.vscode/tasks.json:15]: 💡☕️ I wonder if there is a way to generalise this,

File: src/LocalAgent/Dockerfile_ [hunk 6]
```diff
@@ -32,13 +32,16 @@ ENV MINDARO_BUILD_NUMBER=${MindaroBuildNumber}
...
-RUN chmod +x ./setup-alpine-tools.sh \
-    && ./setup-alpine-tools.sh \
-    && rm -f ./setup-alpine-tools.sh
+RUN apk update
+RUN apk upgrade
+RUN apk add -u bash
+RUN apk add -u procps
+RUN apk add -u curl
+RUN apk add -u bind-tools
+RUN apk add -u kubectl --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing
```
Comment (Tatsinnit) [inline src/LocalAgent/Dockerfile_:44]: 💡☕️ observation: we are adding kubectl, other tools for latter use? And previous setuptool mechanism was doing something different? Just wondering if this is improvement over dotnet6 refactor.

**Review by Tatsinnit — DISMISSED** [review]
❤️💡☕️🎉 Thank you so much for adding me to Thai review, it’s a huge PR and given you have dev tested I am sure all this will rock up awesomely.

I just had few observations to share hence few comments, thank you so much.

File: build/setup-alpine-tools.sh (hunk 4 above)
Comment (hsubramanianaks) [reply build/setup-alpine-tools.sh:1]: For some reason build systems failed due to this. That's why changed them to bash. This is used by devhost agent, restoration agent and routing manager and uploaded as images to ACR.

Comment (hsubramanianaks) [reply build/setup-alpine-tools.sh]: Thanks bro this is taken care.

File: src/LocalAgent/Dockerfile_ (hunk 6 above)
Comment (hsubramanianaks) [reply src/LocalAgent/Dockerfile_]: this is reverted back.

File: src/dsc/Properties/launchSettings.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "profiles": {
+    "dsc": {
+      "commandName": "Project",
+      "commandLineArgs": "connect --service stats-api --control-port 51545 --namespace todo-app --local-port 3001",
+      "environmentVariables": {
+        "BRIDGE_ENVIRONMENT": "dev"
+      },
+      "applicationUrl": "C:\\Users\\hsubramanian\\repos\\forked\\Bridge-To-Kubernetes\\src\\dsc\\bin\\Debug\\net6.0\\win-x64\\publish\\"
```
[the remark this answers is not in the document]
Comment (hsubramanianaks) [reply src/dsc/Properties/launchSettings.json]: removed this file and it is auto generated when debugged via visual studio.

**Review by Tatsinnit — APPROVED** [review]
❤️☕️🎊🚀

**Review by elenavillamil — APPROVED** [review]

**Comment by sabbour** [comment]
Does this close any of those issues?
https://github.com/Azure/Bridge-To-Kubernetes/issues/112
https://github.com/Azure/Bridge-To-Kubernetes/issues/99
https://github.com/Azure/Bridge-To-Kubernetes/issues/98
https://github.com/Azure/Bridge-To-Kubernetes/issues/113

**Comment by hsubramanianaks** [comment]
it is not deployed to production, just in development. so, we can close once it is in production.

**Comment by sabbour** [comment]
Closes issue #42

**Pull request merged**